### PR TITLE
Add a required job checker as an action (1/2)

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

I added the [check-group](https://github.com/tianhaoz95/check-group) code into our Probot in https://github.com/carmocca/probot/commit/d50d1c1be455fcf66fc6deb12e3bd4ec94caba40

This PR adds a GitHub action for it.

Follow-up would be merging https://github.com/Lightning-AI/lightning/pull/14294 after testing.
But the workflow needs to be [in the default branch](https://github.com/Lightning-AI/lightning/pull/14294#discussion_r952804783) to be tested.

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @tchaton @rohitgr7 @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/checkgroup.yml [hunk 1]
```diff
@@ -108,12 +118,10 @@ subprojects:
 
   - id: "lightning_app"
     paths:
-      - ".azure/app-cloud-e2e.yml"
       - "requirements/app/**"
       - "src/lightning_app/**"
       - "tests/tests_app/**"
       - "tests/tests_app_examples/**"
-      - "tests/tests_clusters/**"
```
Comment (carmocca) [inline .github/checkgroup.yml:116]: This test does not run at all: https://github.com/Lightning-AI/lightning/pull/13835#discussion_r950598910

File: .github/checkgroup.yml [hunk 2]
```diff
@@ -108,16 +118,16 @@ subprojects:
...
       # the examples are used in the app CI
       - "examples/app_*"
     checks:
-      - "App.cloud-e2e"
```
Comment (carmocca) [inline .github/checkgroup.yml:120]: This comment: https://github.com/Lightning-AI/lightning/pull/14282#discussion_r950669432 was incorrect

File: .github/workflows/probot-check-group.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Probot
+
+on:
+  check_run: {}
+  pull_request: {}
+  issue_comment: {types: [created]}
```
Comment (carmocca) [inline .github/workflows/probot-check-group.yml:6]: @probot pls

will re-run the action (if necessary)

File: .github/checkgroup.yml [hunk 4]
```diff
@@ -73,7 +84,6 @@ subprojects:
   - id: "pytorch_lightning: Docker"
     paths:
       - "dockers/**"
-      - "!dockers/README.md"
```
Comment (carmocca) [inline .github/checkgroup.yml:76]: Negation is not supported

File: .github/checkgroup.yml [hunk 5]
```diff
@@ -52,14 +46,33 @@ subprojects:
...
+      - "tests/tests_pytorch/run_standalone_*.sh"
+    checks:
+      - "pytorch-lightning (GPUs)"
+
+  - id: "pytorch_lightning: Azure HPU"
+    paths:
+      - ".azure/hpu-tests.yml"
+    checks:
+      - "pytorch-lightning (HPUs)"
+      - "pytorch-lightning (IPUs)"
```
Comment (Borda) [inline .github/checkgroup.yml:68]:
```suggestion
```

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
wizardry!

**Pull request merged**